apps/postgres-new/components/chat.tsx

```diff
@@ -147,7 +162,16 @@ export default function Chat() {
       const [file] = files
 
       if (file) {
-        await sendCsv(file)
+        if (file.type === 'text/csv' || file.name.endsWith('.csv')) {
+          await sendCsv(file)
+        } else if (file.type === 'application/sql' || file.name.endsWith('.sql')) {
+          await sendSql(file)
+        } else {
+          appendMessage({
+            role: 'assistant',
+            content: `Only CSV and SQL files are currently supported.`,
+          })
```

Comment:
For this part I'm sending a raw message to the chat as we don't have a catch all tool which could display a generic error. Is that ok?

Reply:
It's too bad there's no simple way to do a catch-all. We could create a fake tool call for display-only purposes, but I'm not sure if this will confused the model for future requests (since we wouldn't want to define the tool call inside the tool schemas - just within the tool components):

```ts
// Add a fake tool call requesting the file
// with an error indicating the file wasn't supported
appendMessage({
  role: 'assistant',
  content: '',
  toolInvocations: [
    {
      state: 'result',
      toolCallId: generateId(),
      toolName: 'requestFile', // this tool call doesn't actually exist - just as a component
      args: {},
      result: {
        success: false,
        error: `The file has type '${file.type}'. Let the user know that only CSV and SQL imports are currently supported.`,
      },
    },
  ],
})
```

It's probably not worth the hacky-ness, so I think what you have works 👍